algorithmic_efficiency/workloads/ogbg/workload.py

```diff
@@ -81,17 +81,25 @@ def _build_input_queue(self,
 
   # Does NOT apply regularization, which is left to the submitter to do in
   # `update_params`.
-  def loss_fn(self,
-              label_batch: spec.Tensor,
-              logits_batch: spec.Tensor,
-              mask_batch: spec.Tensor,
-              label_smoothing: float = 0.0) -> spec.Tensor:  # differentiable
+  def loss_fn(
+      self,
+      label_batch: spec.Tensor,
+      logits_batch: spec.Tensor,
+      mask_batch: Optional[spec.Tensor] = None,
+      label_smoothing: float = 0.0
+  ) -> Tuple[spec.Tensor, spec.Tensor]:  # differentiable
+    """Return (correct scalar average loss, 1-d array of per-example losses)."""
     per_example_losses = self._binary_cross_entropy_with_mask(
         labels=label_batch,
         logits=logits_batch,
         mask=mask_batch,
         label_smoothing=label_smoothing)
-    return per_example_losses
+    if mask_batch is not None:
+      n_valid_examples = mask_batch.sum()
```

Comment:
for OGBG, the masking is really weird (annoying), where it is actually 2-D because it has per-example and _per-class_ masking, so I think this has to be "number of examples that have a mask that is _all_ zeros" instead of just a sum

Reply:
Hm but this is equivalent to how it was before (see [here](https://github.com/mlcommons/algorithmic-efficiency/blob/main/reference_algorithms/development_algorithms/ogbg/ogbg_pytorch/submission.py#L60-L62)), unless I'm missing something. Or was it wrong before already?

Reply:
Hm but this is equivalent to how it was before (see [here](https://github.com/mlcommons/algorithmic-efficiency/blob/main/reference_algorithms/development_algorithms/ogbg/ogbg_pytorch/submission.py#L60-L62)), unless I'm missing something. Or was it wrong before already?

Reply:
ok, I think I confused myslef, and this is the way that we do the padding in init2winit. I think you're right because this is basically doing a per-class sigmoid_binary_cross_entropy calculation, and we want to mask out some classes too, so it's valid to sum across the whole mask. sorry for the confusion!!